tree/ntuple/v7/src/RNTupleJoinTable.cxx

```diff
@@ -84,28 +56,47 @@ void ROOT::Experimental::Internal::RNTupleJoinTable::Build()
                                                                 "std::int64_t",  "std::uint8_t", "std::uint16_t",
                                                                 "std::uint32_t", "std::uint64_t"};
 
+   pageSource.Attach();
+   auto desc = pageSource.GetSharedDescriptorGuard();
+
+   fJoinFieldValueSizes.reserve(fJoinFieldNames.size());
+   std::vector<std::unique_ptr<RFieldBase>> fields;
    std::vector<RFieldBase::RValue> fieldValues;
-   fieldValues.reserve(fJoinFields.size());
+   fieldValues.reserve(fJoinFieldNames.size());
+
+   for (const auto &fieldName : fJoinFieldNames) {
+      auto fieldId = desc->FindFieldId(fieldName);
+      if (fieldId == ROOT::kInvalidDescriptorId)
+         throw RException(R__FAIL("could not find join field \"" + std::string(fieldName) + "\" in RNTuple \"" +
+                                  pageSource.GetNTupleName() + "\""));
```

Comment:
Add information on the storage location of the RNTuple perhaps? e.g. if this is building from a chain of ntuples the name could be the same for every one

Reply:
Unfortunately `RPageSource` does not provide this information, so this is something that will need to be handled by the upstream application (i.e., the `RNTupleProcessor`). `Build` could return a `RResult` instead to propagate the error instead of throwing it directly, but since the `RNTupleJoinTable` as well as its integration into the `RNTupleProcessor` will still change significantly I would propose to address this at a later point.